File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n<=1) {
            return 1;
        }
        
        int[] fn = new int[n + 1];
        fn[1] = 1;
        fn[2] = 2;

        for(int i=3;i<=n;i++){
            fn[i]=fn[i-1]+fn[i-2];
        }
        
        return fn[n];
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n):
        if n==0: return 0
        if n==1: return 1
        if n==2: return 2
        dp = [0]*(n+1) 
        dp[1]= 1
        dp[2] = 2
        for i in range(3,n+1):
            dp[i] = dp[i-1] +dp[i-2]
        print(dp)
        return dp[n]